client/task-list/tasks/payments/RemotePayments/components/PaymentConnect.js

```diff
@@ -2,58 +2,69 @@
  * External dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
-import { Button } from '@wordpress/components';
 import interpolateComponents from 'interpolate-components';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { Form, Link, TextControl } from '@woocommerce/components';
+import {
+	Link,
+	Settings,
+	WooRemotePaymentSettings,
+	Spinner,
+} from '@woocommerce/components';
+import apiFetch from '@wordpress/api-fetch';
+import { useEffect, useState } from '@wordpress/element';
 import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { useSlot, Text } from '@woocommerce/experimental';
 
 export const PaymentConnect = ( {
 	markConfigured,
 	method,
 	recordConnectStartEvent,
 } ) => {
-	const { api_details_url: apiDetailsUrl, fields, key, title } = method;
+	const {
+		api_details_url: apiDetailsUrl,
+		fields: fieldsConfig,
+		key,
+		title,
+	} = method;
+
 	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
 	const { createNotice } = useDispatch( 'core/notices' );
+	const slot = useSlot( `woocommerce_remote_payment_settings_${ key }` );
+	const hasFills = Boolean( slot.fills && slot.fills.length );
+	const [ state, setState ] = useState( 'loading' );
+	const [ fields, setFields ] = useState( null );
+
+	// This transform will be obsolete when we can derive essential fields from the API
+	const settingsTransform = ( settings ) => {
+		const essentialFields = fieldsConfig.map( ( field ) => field.name );
+
+		return Object.values( settings ).filter( ( setting ) =>
+			essentialFields.includes( setting.id )
+		);
+	};
+
+	useEffect( () => {
+		apiFetch( {
+			path: `/wc/v3/payment_gateways/${ key }/`,
+		} )
+			.then( ( results ) => {
+				setFields( settingsTransform( results.settings ) );
+				setState( 'loaded' );
+			} )
+			.catch( ( e ) => {
+				setState( 'error' );
+				/* eslint-disable no-console */
+				console.error( e );
+				/* eslint-enable no-console */
+			} );
+	}, [] );
```

Comment:
Yup, there are a few small changes I'll need to address after some pending PRs go through. Right now I'm keen to merge things and iterate instead of having several PRs pile up waiting for eachother.